tests/plugins/autocomplete/autocomplete.js

```diff
@@ -97,6 +97,99 @@
 			ac.destroy();
 		},
 
+		'test tab inserts match': function() {
+			var editor = this.editor, bot = this.editorBot, editable = editor.editable(),
+				ac = new CKEDITOR.plugins.autocomplete( this.editor, matchTestCallback, dataCallback );
+
+			bot.setHtmlWithSelection( '' );
+
+			editor.editable().fire( 'keyup', new CKEDITOR.dom.event( {} ) );
+
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 40 } ) ); // ARROW DOWN
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 9 } ) ); // TAB
+
+			assert.areEqual( '<p>item1</p>', editor.getData() );
+
+			ac.destroy();
+		},
+
+		'test custom single key inserts match': function() {
+			var editor = this.editor, bot = this.editorBot, editable = editor.editable(),
+				ac = new CKEDITOR.plugins.autocomplete( this.editor, matchTestCallback, dataCallback );
+
+			bot.setHtmlWithSelection( '' );
+
+			ac.commitKeystroke = 16; // SHIFT
+
+			editor.editable().fire( 'keyup', new CKEDITOR.dom.event( {} ) );
+
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 40 } ) ); // ARROW DOWN
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 16 } ) ); // SHIFT
+
+			assert.areEqual( '<p>item1</p>', editor.getData() );
+
+			ac.destroy();
+		},
+
+		'test custom keys inserts match': function() {
+			var editor = this.editor, bot = this.editorBot, editable = editor.editable(),
+				ac = new CKEDITOR.plugins.autocomplete( this.editor, matchTestCallback, dataCallback );
+
+			bot.setHtmlWithSelection( '' );
+
+			ac.commitKeystroke.push( 16 ); // SHIFT
+
+			editor.editable().fire( 'keyup', new CKEDITOR.dom.event( {} ) );
+
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 40 } ) ); // ARROW DOWN
+			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 16 } ) ); // SHIFT
+
+			assert.areEqual( '<p>item1</p>', editor.getData() );
+
+			ac.destroy();
+		},
+
+		'test custom keys inserts match (global configuration)': function() {
+			var editor = this.editor, bot = this.editorBot, editable = editor.editable(),
+				configKeyCodes = CKEDITOR.config.autocomplete_commitKeystroke.slice();
```

Comment:
Instead of trickery like that, just use a separate editor for that.

Make one "clean" editor that will be used in the tests that are already created, and add new editor(s) for your custom config. See [Easy Image tests](https://github.com/ckeditor/ckeditor-dev/blob/82801e3/tests/plugins/easyimage/config.js) as an example.

That solution will be much cleaner.